gemrb/core/DisplayMessage.cpp

```diff
@@ -187,30 +188,63 @@ void DisplayMessage::DisplayString(String text, const Color &color, Scriptable *
 	}
 }
 
+std::vector<std::string> DisplayMessage::GetAllColors() const
+{
+	std::vector<std::string> auxiliaryColors;
+	AutoTable colorTable = gamedata->LoadTable("colors", true);
+	assert(colorTable);
+	for (TableMgr::index_t r = 0; r < colorTable->GetRowCount(); r++) {
```

Comment:
This is missing a call to `reserve` which will lead to a whole bunch of allocations
```suggestion
        auxiliaryColors.reserve(colorTable->GetRowCount());
	for (TableMgr::index_t r = 0; r < colorTable->GetRowCount(); r++) {
```